spec/addressable/template_spec.rb

```diff
@@ -728,6 +728,21 @@ def self.match(name)
 end
 
 describe Addressable::Template do
+  describe 'initialize' do
+    context 'with a non-string' do
+      it 'raises a TypeError' do
+        expect { Addressable::Template.new(nil) }.to raise_error(TypeError)
+      end
+    end
+  end
+
+  describe 'freeze' do
```

Comment:
This is the only template spec that appears to be related to freezing a template. This should be the only template spec in this PR.